be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -335,10 +309,15 @@ Status SinkBuffer::_try_to_send_rpc(const TUniqueId& instance_id, const std::fun
                     return Status::OK();
                 }
                 // this is the last eos query, set query stats
-                _eos_query_stats[instance_id.lo]->merge_pb(request.params->query_statistics());
-                request.params->clear_query_statistics();
-                _eos_query_stats[instance_id.lo]->to_pb(request.params->mutable_query_statistics());
-                _eos_query_stats[instance_id.lo]->clear();
+                if (auto final_stats = _fragment_ctx->runtime_state()->query_ctx()->intermediate_query_statistic()) {
+                    final_stats->to_pb(request.params->mutable_query_statistics());
+                }
+            }
+        } else {
+            if ((_request_sent & _sent_audit_stats_frequency) == _sent_audit_stats_frequency) {
```

Comment:
What if total `_request_sent` is less than 127 ?

Reply:
that will be up the eos's last request.